### PR TITLE
feat: Fix SSL support, make SSL default feature

### PR DESCRIPTION
**Pull request by untitaker**
Usage of tls_listen_addr (running a relay capable of serving HTTPS
responses) might be not too popular of a usecase, but speaking SSL to an
upstream relay certainly is if you're using ingest.sentry.io (i.e.
you're not fully onprem)

- [ ] test using gobetween's ssl feature -- DELAYED, see jira

### CODE REVIEW

**Review by mitsuhiko — APPROVED** [review]

**Comment by untitaker** [comment]
I am concerned about ease of use on osx and especially Windows when openssl becomes mandatory. Is there any reason not to use native-tls?

**Comment by mitsuhiko** [comment]
It native tls works no reason not to use it. Pretty sure originally it did not, at least not when http2 was in the mix.

**Comment by untitaker** [comment]
Yup, still the same. Forgot about h2. But there does appear to be a way to get HTTPS with HTTP1 using native-tls, it's just not documented.

**Comment by untitaker** [comment]
Updated: http2 support is (probably) gone, SSL works fine in both directions.

The fact that we now need the private key + cert in a different format now is not pretty.

**Comment by untitaker** [comment]
@mitsuhiko please re-review

**Comment by untitaker** [comment]
tests blocked on https://github.com/actix/actix-web/pull/491

File: server/Cargo.toml [hunk 1]
```diff
@@ -10,8 +10,8 @@ version = "0.1.3"
 build = "build.rs"
 
 [features]
-default = []
-with_ssl = ["openssl", "actix-web/alpn"]
+default = ["with_ssl"]
+with_ssl = ["native-tls", "actix-web/tls"]
```
Comment (jan-auer) [inline server/Cargo.toml:14]: Can we please call this feature `"ssl"`? See https://rust-lang-nursery.github.io/api-guidelines/naming.html#c-feature

**Review by jan-auer — CHANGES_REQUESTED** [review]

File: server/Cargo.toml (hunk 1 above)
Comment (untitaker) [reply server/Cargo.toml:14]: done, forgot to push

**Review by jan-auer — APPROVED** [review]

File: server/Cargo.toml (hunk 1 above)
Comment (jan-auer) [reply server/Cargo.toml:14]: _Offline conversation:_ we're not doing this.

**Comment by jan-auer** [comment]
@untitaker This is green now, g2g.

**Pull request merged**